controllers/platform/services/properties.go

```diff
@@ -185,19 +186,21 @@ func GenerateDataIndexWorkflowProperties(workflow *operatorapi.SonataFlow, platf
 // Never nil.
 func GenerateJobServiceWorkflowProperties(workflow *operatorapi.SonataFlow, platform *operatorapi.SonataFlowPlatform) (*properties.Properties, error) {
 	props := properties.NewProperties()
-	props.Set(constants.JobServiceRequestEventsConnector, constants.QuarkusHTTP)
-	props.Set(constants.JobServiceRequestEventsURL, fmt.Sprintf("%s://localhost/v2/jobs/events", constants.JobServiceURLProtocol))
 	js := NewJobServiceHandler(platform)
-	if workflow != nil && !profiles.IsDevProfile(workflow) && js.IsServiceEnabled() {
-		if workflowdef.HasTimeouts(workflow) {
-			props.Set(constants.KogitoJobServiceHealthCheckEnabled, "true")
-		}
-		p, err := js.GenerateWorkflowProperties()
-		if err != nil {
-			return nil, err
+	if !profiles.IsDevProfile(workflow) && workflow != nil && workflow.Status.Services != nil && workflow.Status.Services.JobServiceRef != nil {
+		serviceBaseUrl := workflow.Status.Services.JobServiceRef.Url
```

Comment:
I see something that I believe more difficult to follow now, or potentially error prone. Let me explain:

**Before:**
    The jobs-service url was calculated by the JobServiceHandler, and there, it was derived somehow directly from that service, or, from the cluster platform if any.

**Now:**
    Get get the jobs-service url from the  workflow.Status.Services.JobServiceRef, BUT, the workflow is normally what is being deployed at this point, and that somehow "caused" the execution of all this cycle. So we configure some WF properties from values from the WF itself.

I am correct?


Same reasoning apply for the DI

Reply:
no, we are still using the service handlers to set the urls. There's a new method called `SetServiceUrlsInWorkflowStatus` that handles this in the workflow(s) status -
https://github.com/apache/incubator-kie-kogito-serverless-operator/blob/2325390963e643afd9868174e9818d7dafde4317/controllers/platform/services/services.go#L432-L439

what HAS changed is... during a workflow's configmap properties creation, instead of calling the handler directly, we're using the service url directly from the workflow's status... which has already been properly set by the service handlers.

this is actually LESS error prone because it ensures the accuracy of some things, which, in `main` today can be a bit nebulous -

1. this ensures that a workflow's application properties match the correct service urls which are set the status of a workflow. currently a user would have to figure this out for themselves by either checking the config map, or the platform object.
2. this ensures that application properties config maps are properly updated in real-time when service changes are made in the platform/clusterplatform... which, in turn, forces the workflow pod to restart and these new settings to take affect. today, a user may have to remove the existing configmap to force proper properties... which is problematic because they may not know this is required and a workflow may be attempting to send to a non-existent (or incorrect) service.

Reply:
its important to keep in mind that, workflows are re-reconciled anytime a related platform object is changed -
https://github.com/apache/incubator-kie-kogito-serverless-operator/blob/fff6361bde07b566a22bc24ec7be84506e7b98ae/controllers/sonataflow_controller.go#L174-L181

this ensures that service-related changes at the platform level will result in a workflow.status change... which will result in a configmap change and workflow pod restart. this has all been thoroughly tested.

Reply:
one more note on this. this is the final part of a long planned, multi-tasked, implementation of supporting services in the operator. `workflow.status.services` has not existed until now, so we previously had no choice but to call the service handlers directly during workflow properties creation. with this PR, we'll have reached the intended "final" implementation.

Reply:
Ok, many thanks for the explanation.
The part that wasn't clear to me was "during a workflow's configmap properties creation, instead of calling the handler directly, we're using the service url directly from **the workflow's status... which has already been properly set by the service handlers.**"